src/functionalTest/java/uk/gov/hmcts/sptribs/caseworker/CaseworkerCancelHearingFT.java

```diff
@@ -1,77 +1,119 @@
     }
 
     @Test
-    public void shouldReceiveNoticeWhenSubmittedIsInvoked() throws Exception {
-        final Map<String, Object> caseData = caseData(REQUEST_SUBMITTED);
+    public void shouldReceiveNoticeWhenOnePartyIsSubmittedIsInvoked() throws Exception {
+        final Map<String, Object> caseData = caseData(REQUEST_SUBMITTED_ONE_PARTY);
 
         final Response response = triggerCallback(caseData, EventConstants.CASEWORKER_CANCEL_HEARING, SUBMITTED_URL);
+
         assertThat(response.getStatusCode()).isEqualTo(OK.value());
+        assertThatJson(response.asString())
+            .inPath(CONFIRMATION_HEADER)
+            .isEqualTo("# Hearing cancelled \\n## A notification has been sent to: Subject");
+    }
+
+    @Test
+    public void shouldReceiveErrorIfInvalidRequestIsInvoked() throws Exception {
+        JSONObject emptyJsonObject = new JSONObject();
+        Map<String, Object> caseData = new HashMap<>();
+        caseData.put("emptyString", "");
+        caseData.put("emptyJsonObject", emptyJsonObject);
+
+        final Response response = triggerCallback(caseData, EventConstants.CASEWORKER_CANCEL_HEARING, SUBMITTED_URL);
 
-        // notes.date value is compared using ${json-unit.any-string}
-        // assertion will fail if the above value is missing
         assertThatJson(response.asString())
-            .when(IGNORING_EXTRA_FIELDS)
-            .isEqualTo(json(expectedResponse(
-                "classpath:responses/response-caseworker-cancel-hearing-submitted.json"
-            )));
+            .inPath(CONFIRMATION_HEADER)
+            .isEqualTo("# Cancel hearing notification failed \\n## Please resend the notification");
     }
 
     @Test
     public void shouldRaiseErrorIfSubjectRepresentativeRespondentNotifyPartiesAllNull() throws Exception {
-        Map<String, Object> caseData = caseData(REQUEST_MID_EVENT);
+        Map<String, Object> caseData = caseData(REQUEST_ALL_PARTIES_NULL);
 
         Response response = triggerCallback(caseData, EventConstants.CASEWORKER_CANCEL_HEARING, RECORD_NOTIFY_PARTIES_MID_EVENT_URL);
 
         assertThat(response.getStatusCode()).isEqualTo(OK.value());
+        assertThatJson(response.asString())
+            .inPath(ERROR_MESSAGE)
+            .isEqualTo("[\"One recipient must be selected.\"]");
+    }
+
+    @Test
+    public void shouldRaiseErrorIfOnePartyIsInvalid() throws Exception {
+        Map<String, Object> caseData = caseData(REQUEST_ONE_PARTY_INVALID);
 
+        Response response = triggerCallback(caseData, EventConstants.CASEWORKER_CANCEL_HEARING, RECORD_NOTIFY_PARTIES_MID_EVENT_URL);
+
+        assertThat(response.getStatusCode()).isEqualTo(OK.value());
         assertThatJson(response.asString())
-            .when(IGNORING_EXTRA_FIELDS)
-            .isEqualTo(json(expectedResponse(
-                "classpath:responses/response-caseworker-cancel-hearing-record-notify-parties-mid-event.json"
-            )));
+            .inPath(ERROR_MESSAGE)
```

Comment:
since errors is an array, you can use isArray() like below
`assertThatJson(response.asString())
            .inPath("$.errors")
            .isArray()
            .contains("One recipient must be selected.")`

Reply:
Same applies to all other tests above